modules/lily-compiler/src/main/java/io/github/tomboyo/lily/compiler/icg/AstGenerator.java

```diff
@@ -27,7 +29,30 @@ public static Stream<Ast> evaluate(PackageName basePackage, OpenAPI openAPI) {
   }
 
   private Stream<Ast> evaluate(OpenAPI openAPI) {
-    return Stream.of(evaluateComponents(openAPI), evaluatePaths(openAPI)).flatMap(identity());
+    var table =
+        Stream.of(evaluateComponents(openAPI), evaluatePaths(openAPI))
+            .flatMap(identity())
+            /*
+             * Collect definitions into a FQN -> AST table. This gives us an
+             * opportunity to modify AST generated from schema encountered early based
+             * on schema encountered late. We have to do this to create sealed
+             * interfaces, for example, which reference and are referenced by their
+             * permitted members.
+             */
+            .collect(
+                Collectors.toMap(
+                    ast -> ast.name(),
+                    ast -> ast,
+                    (left, right) -> {
+                      if (left instanceof Modifier mod && right instanceof Definition def) {
+                        return mod.modify(def);
+                      } else if (left instanceof Definition def && right instanceof Modifier mod) {
+                        return mod.modify(def);
+                      } else {
+                        throw new IllegalStateException("Tried to assign two AST to one FQN");
```

Comment:
We need to update this. Though this kind of thing Should Not Happen (tm), when it does we don't want to completely block the user from using the schema, especially since other parts of it may have generated useful code. There are three cases we should think about:

1. Left and right are both a definition (name conflict).
2. Left and right are both modifiers (e.g. when a class is a member of two sealed interfaces). Modifiers should compose (something something they are just functions with extra steps)
3. Left is a modifier that cannot modify right (or vice versa).

We may want to add a general rule to this project that exceptions are bad and we don't want to throw them.

Reply:
Related to https://github.com/Tomboyo/lily/issues/86

Reply:
An interesting problem: if we choose to render one class and discard the other, we'd want to make that choice _consistently_ between builds so that we don't break someone's build. And, if we discard a class, that could break a sealed interface declaration somewhere else.

I'm wondering if maybe failing here and now is actually the right thing to do, and we need to ask the end user how to fix the issue. I don't want users to modify an OAS document locally, but giving them an "override" file where they can write "rules" that say how to resolve _specific_ conflicts could work. It would let them consume from the source of truth, keep their build stable, and make progress locally while waiting on upstream changes. This would have to be its own ticket and could potentially be a path for solving all manner of invalid specification issues (Like name conflicts!)

Reply:
We'll resolve in issue #92.